packages/flutter_test/lib/src/controller.dart

```diff
@@ -699,6 +699,47 @@ abstract class WidgetController {
   ///
   /// Shorthand for `Scrollable.ensureVisible(element(finder))`
   Future<void> ensureVisible(Finder finder) => Scrollable.ensureVisible(element(finder));
+
+  /// Repeatedly scrolls the `scrollable` by `dScroll` until `finder` and
+  /// throws if `finder` is not found for maximum `timeout` times.
+  ///
+  /// Between each scroll, wait for `duration` time for settling.
+  ///
+  /// This is different from [ensureVisible] in that this allows looking for
+  /// `finder` that is not built yet, but requires to specify the scrollable.
+  ///
+  /// If the `scrollable` is infinite and `dScroll`
+  Future<void> scrollUntilVisible(
+    Finder finder,
+    Finder scrollable,
+    double dScroll, {
+      int timeout = 50,
+      Duration duration = const Duration(milliseconds: 50),
+    }
+  ) async {
+    assert(timeout > 0);
+    Offset moveStep;
+    switch(widget<Scrollable>(scrollable).axisDirection) {
```

Comment:
We should handle null here - I imagine we'd want to throw if it's null, but not throw like this.

Reply:
Not sure which `null` you mean but I added some `assert`. Let me know if I misunderstood it.

Reply:
We should assert that widget(scrollable) does not return null before using it.

Reply:
`assert(scrollable.evaluate().isNotEmpty);` should do the job. And according to `WidgetController.widget` implementation  (https://api.flutter.dev/flutter/flutter_test/WidgetController/widget.html) this will not give `null` if there isn't such a widget but throw.

Reply:
The finder isn't necessarily of that type

Reply:
And if your concern is `Element.widget`, `assert(widget != null)` is in its constructor.

Reply:
> The finder isn't necessarily of that type

If the type is wrong, it won't give `null` either. Than would I should do is `assert(finder.evaluate().single.widget is Scrollable)`.

Reply:
Now that I had a look at other APIs, non of them are doing `assert(finder != null)` or `assert(finder.evaluate().isNotEmpty)`. Probably I shouldn't do this either. @dnfield what do you think?

Reply:
removed those assert.

Reply:
The cases when we do assert != null is when getting renderObject from element, which makes sense because along widget->element->renderObject, we can have prior exists without latter, but not vice versa

Reply:
Ahh ok, I see now. This would throw a cast error if it was the wrong type of widget already, not return null.